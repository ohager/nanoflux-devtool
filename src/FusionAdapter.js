import Nanoflux from 'nanoflux-fusion/src/nanoflux-fusion';
import BaseStoreAdapter from './BaseStoreAdapter';

class FusionAdapter extends BaseStoreAdapter {

	mount(context, updateCallback) {
		this.listener = Nanoflux.getFusionStore().subscribe(context, updateCallback);
		console.log("Fusion Store mounted");
	}

	unmount() {
		this.listener.unsubscribe();
		console.log("Fusion Store unmounted");
	}

	getState() {
		return Nanoflux.getFusionStore().getState()
	}

}

export default FusionAdapter;
