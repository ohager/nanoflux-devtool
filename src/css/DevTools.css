.devtools-container h2{
	text-align: center;
}

.devtools-container {
	text-align: left;
	position: absolute;
	top: 0;
	background-color: #111;
	color: #efefef;
	opacity: 0.95;
	min-height: 100%;
	height: 100%;
	width: 240px;
	box-shadow: 5px 0px 2px 0px rgba(2, 2, 2, 0.9);
}