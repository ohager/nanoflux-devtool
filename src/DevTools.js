import React, {Component} from 'react';

import './css/DevTools.css';

const DevToolsStateItem  = (props) =>
		(
			<div>
				<p>{JSON.stringify(props.storeState)}</p>
			</div>
		)

DevToolsStateItem.propTypes = {
	storeState : React.PropTypes.object
};

class DevTools extends Component {

	constructor() {
		super();
		this.state = { };
	}

	onStoreUpdate(state){
		this.setState( {storeState: state });
	}

	componentWillMount(){
		this.props.storeAdapter.mount(this, this.onStoreUpdate);
	}

	componentWillUnmount(){
		this.props.storeAdapter.unmount();
	}

	renderStateItems(){
		//const stateProps = Object.keys(this.state.storeState);
		//return stateProps.map( (state, index) => <DevToolsStateItem key={index}/>)
		return <DevToolsStateItem storeState={this.state.storeState}/>
	}

	render() {
		return (
			<div className="devtools-container">
				<h2>Flux DevTools</h2>
				<hr/>
				{ this.renderStateItems() }
			</div>
		);
	}
}

DevTools.propTypes = {
	storeAdapter : React.PropTypes.object.isRequired
};

export default DevTools;