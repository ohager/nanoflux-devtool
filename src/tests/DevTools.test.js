import React from 'react';
import Renderer from 'react-test-renderer';
import DevTools from '../DevTools'
import FusionAdapter from '../FusionAdapter'

it('Renders initial DevTools', () => {
  const tree = Renderer.create(<DevTools storeAdapter={new FusionAdapter()}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
