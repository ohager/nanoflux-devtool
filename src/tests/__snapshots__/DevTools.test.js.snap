exports[`test Renders initial DevTools 1`] = `
<div
  className="devtools-container">
  <h2>
    Flux DevTools
  </h2>
  <hr />
  <div>
    <p />
  </div>
</div>
`;
